package login;

/////////////////////////////////////////////////////////////协议号///////////////////////////////////////////////////////////////////
enum LoginProtocol
{
	REQ_LOGIN			= 1000;					// 请求登陆
	REQ_REGISTER		= 1001;					// 请求注册
}

/////////////////////////////////////////////////////////////错误码///////////////////////////////////////////////////////////////////
enum LoginError
{
	LOGIN_SUCCESS 		= 0;					// 登陆成功
	USER_NOT_EXIST 		= 1;					// 用户不存在
	PASSWD_ERROR 		= 2;					// 密码错误
	USER_HAS_REGISTER 	= 3;					// 用户已经注册	
}

/////////////////////////////////////////////////////////////协议///////////////////////////////////////////////////////////////////
message LoginReq
{
	optional string userName 	= 1;		// 用户名
	optional string passwd		= 2;		// 密码
}

message LoginAck
{
	optional uint32 errorCode	= 1;		// 错误码
	optional uint32 loginToken	= 2;		// 登陆成功返回的token
}

message RegisterReq
{
	optional string userName	= 1;		// 用户名
	optional string passwd		= 2;		// 密码
}

message RegisterAck
{
	optional uint32 errorCode	= 1;		// 错误码
}